### DISCUSSION TITLE
TRPC Dynamic Procedure URL

### DISCUSSION THREAD

**Discussion by rasel-stacklearner** [trpc/trpc#5626] [Q&A]
I am currently using QStash as a message queue with TRPC. QStash is HTTP-based and in the example below, you can see that I call the TRPC procedure URL when publishing a message using QStash. My question is, how can I make this URL dynamic? Instead of repeatedly writing the full URL, I want to use a function call like 
js 
trpc.subscription.updateSubscription.url
.
how can I do this in trpc?

![image](https://github.com/trpc/trpc/assets/141147182/4c200d02-922a-4ca7-b380-1c2c6547550a)

**Comment by conduongtong** [comment]
You mean this: https://trpc.io/docs/rpc#accessing-nested-procedures?

**Reply by rasel-stacklearner** [reply]
I am not referring to nested procedures. I am merely seeking a way to obtain the procedure path from the client application. I require a solution where I can simply provide the reference of the TRPC procedure, such as =>  trpc.course.courseList, and it will return the path, such as "http://localhost:3000/api/trpc/course.courseList". I want to achieve this because it will ensure type safety.
